### ISSUE TITLE
Blacklisting a package that contains "jn" in its name gets "jn" added to make args

### ISSUE THREAD

**Issue by haudren** [catkin/catkin_tools#555]
### System Info

* Operating System: Ubuntu Bionic Beaver (18.04) x86_64
* Python Version: Python 2.7.15+
* Version of catkin_tools: 0.4.5
* ROS Distro: melodic

### Expected Behavior

Configuring a workspace by blacklisting a package, whatever its name, should not affect make-args. 

### Actual Behavior

Configuring a workspace by blacklisting a package starting by jn adds it to the make args, which potentially slows down the build.

### Steps to Reproduce the Issue

```bash
mkdir -p test_ws/src
cd test_ws/src
catkin_create_pkg j2pkg
cd ..
catkin build
catkin config --blacklist j2pkg
```

Observe the result:
```
-------------------------------------------------
Additional CMake Args:       None
Additional Make Args:        j2
Additional catkin Make Args: None
Internal Make Job Server:    True
Cache Job Environments:      False
-------------------------------------------------
```

**Comment by gavanderhoorn** [comment]
Does `--blacklist 'j2pkg'` change how the args are interpreted?

**Comment by haudren** [comment]
Hey sorry for the late reply: no it doesn't. Using both `'j2pkg'` and `"j2pkg"` results in the same behaviour.

**Comment by jarvisschultz** [comment]
Probably a closely-related issue is trying to build a single package or list of packages that contain `jn` in their names. The `jn` is automatically consumed by the `make-args`, and stripped from the package list:

```
catkin build j2example_msgs
-----------------------------------------------------------
Profile:                     default
Extending:        [explicit] /opt/ros/kinetic
Workspace:                   /home/jarvis/examplews
-----------------------------------------------------------
Build Space:        [exists] /home/jarvis/examplews/build
Devel Space:        [exists] /home/jarvis/examplews/devel
Install Space:      [unused] /home/jarvis/examplews/install
Log Space:          [exists] /home/jarvis/examplews/logs
Source Space:       [exists] /home/jarvis/examplews/src
DESTDIR:            [unused] None
-----------------------------------------------------------
Devel Space Layout:          linked
Install Space Layout:        None
-----------------------------------------------------------
Additional CMake Args:       None
Additional Make Args:        j2
Additional catkin Make Args: None
Internal Make Job Server:    True
Cache Job Environments:      False
-----------------------------------------------------------
Whitelisted Packages:        None
Blacklisted Packages:        None
-----------------------------------------------------------
Workspace configuration appears valid.
-----------------------------------------------------------
[build] Given package 'example_msgs' is not in the workspace
```

**Comment by timonegk** [comment]
This was fixed in #573.
